README.md

```diff
@@ -137,6 +137,255 @@ To download file you should call **GetFile** method
+			  client = new TelegramClient(APIId, APIHash, session);
+			  // subscribe an event to receive live messages
+			  client.Updates += Client_Updates;
+			  await client.ConnectAsync();
+			  Console.WriteLine($"Authorised: {client.IsUserAuthorized()}");
+			  TLUser user = null;
+			  // -- If the user has already authenticated, this step will prevent account from being blocked as it
+			  // -- reuses the data from last authorisation.
+			  if (client.IsUserAuthorized())
+				  user = client.Session.TLUser;
+			  else
+			  {
+				  var registered = await client.IsPhoneRegisteredAsync(phone);
+				  var hash = await client.SendCodeRequestAsync(phone);
+				  Console.Write("Code: ");
+				  var code = Console.ReadLine();
+				  if (!registered)
+				  {
+					  Console.WriteLine($"Sign up {phone}");
+					  user = await client.SignUpAsync(phone, hash, code, "First", "Last");
+				  }
+				  Console.WriteLine($"Sign in {phone}");
+				  user = await client.MakeAuthAsync(phone, hash, code);
+			  }
+
+			  var contacts = await client.GetContactsAsync();
+			  Console.WriteLine("Contacts:");
+			  foreach (var contact in contacts.Users.OfType<TLUser>())
+			  {
+				  var contactUser = contact as TLUser;
+				  Console.WriteLine($"\t{contact.Id} {contact.Phone} {contact.FirstName} {contact.LastName}");
+			  }
+
+
+			  var dialogs = (TLDialogs) await client.GetUserDialogsAsync();
+			  Console.WriteLine("Channels: ");
+			  foreach (var channelObj in dialogs.Chats.OfType<TLChannel>())
+			  {
+				  var channel = channelObj as TLChannel;
+				  Console.WriteLine($"\tChat: {channel.Title}");
+			  }
+
+			  Console.WriteLine("Groups:");
+			  TLChat chat = null;
+			  foreach (var chatObj in dialogs.Chats.OfType<TLChat>())
+			  {
+				  chat = chatObj as TLChat;
+				  Console.WriteLine($"Chat name: {chat.Title}");
+				  var request = new TLRequestGetFullChat() { ChatId = chat.Id };
+				  var fullChat = await client.SendRequestAsync<TeleSharp.TL.Messages.TLChatFull>(request);
+
+				  var participants = (fullChat.FullChat as TeleSharp.TL.TLChatFull).Participants as TLChatParticipants;
+				  foreach (var p in participants.Participants)
+				  {
+					  if (p is TLChatParticipant)
+					  {
+						  var participant = p as TLChatParticipant;
+						  Console.WriteLine($"\t{participant.UserId}");
+					  }
+					  else if (p is TLChatParticipantAdmin)
+					  {
+						  var participant = p as TLChatParticipantAdmin;
+						  Console.WriteLine($"\t{participant.UserId}**");
+					  }
+					  else if (p is TLChatParticipantCreator)
+					  {
+						  var participant = p as TLChatParticipantCreator;
+						  Console.WriteLine($"\t{participant.UserId}**");
+					  }
+				  }
+
+				  var peer = new TLInputPeerChat() { ChatId = chat.Id };
+				  var m = await client.GetHistoryAsync(peer, 0, 0, 0);
+				  Console.WriteLine(m);
+				  if (m is TLMessages)
+				  {
+					  var messages = m as TLMessages;
```

Comment:
to avoid the use of `is` and `as` afterwards, you can do `(if Messages m is TLMessages)` I think (new feature of C#)

Reply:
like that :-)